File: frontend/src/services/MolecularActivityService.js
import axios from "../custom-axios/axios"

const MolecularActivityService={
    getMolecularActivityAbstract:()=>{
        return axios.get('/molecularActivity')
    },
    getDetailsAboutMolecularActivity: (name)=>{
        return axios.get(`/molecularActivity/${name}`)
    }
}

export default MolecularActivityService

File: backend/src/main/java/com/example/backend/service/MolecularActivityService.java
package com.example.backend.service;

import com.example.backend.model.MolecularActivity;
import com.example.backend.model.exceptions.MolecularActivityDoesNotExist;

import java.io.IOException;
import java.util.Optional;

public interface MolecularActivityService {
    Optional<MolecularActivity> getDetailsAboutMolecularActivity(String molecularActivityName) throws IOException, InterruptedException, MolecularActivityDoesNotExist;

    String getMolecularActivityAbstract();
}
